addon/-private/system/record-arrays/record-array.js

```diff
@@ -201,9 +202,16 @@ export default Ember.ArrayProxy.extend(Ember.Evented, {
     this._super.apply(this, arguments);
   },
 
-  createSnapshot(options) {
-    var adapterOptions = options && options.adapterOptions;
-    var meta = this.get('meta');
-    return new SnapshotRecordArray(this, meta, adapterOptions);
+  createSnapshot(options = {}) {
+    const adapterOptions = options.adapterOptions;
+    const meta = this.get('meta');
+
+    const snapshot = new SnapshotRecordArray(this, meta, adapterOptions);
```

Comment:
Go ahead a change it to improve the consistency. The Snapshot constructors are private.